internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,507 @@
+) []*iammodel.VmwareTanzuCoreV1alpha1PolicySubject {
+	var newList []*iammodel.VmwareTanzuCoreV1alpha1PolicySubject
+	// nested iteration for preserving order of subjects.
+	for _, each := range state {
+		for _, sub := range server {
+			if sub.Name == each.Name && *sub.Kind == *each.Kind {
+				newList = append(newList, each)
+			}
+		}
+	}
+
+	return newList
+}
+
+func constructRBOpForUpdate(rbl []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding, subjectIntersect *map[string]iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType, setAction iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType) {
+	var (
+		flipAction    iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType
+		toBeFlippedTo iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType
+	)
+
+	if setAction == iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpTypeADD {
+		flipAction = iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpTypeDELETE
+		toBeFlippedTo = iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpTypeOPTYPEUNSPECIFIED
+	}
+
+	for _, rb := range rbl {
+		for _, sub := range rb.Subjects {
+			key := helper.AddDelimiterToItems(roleSubjectDelimiter, rb.Role, sub.Name, string(*sub.Kind))
+			if action, ok := (*subjectIntersect)[key]; ok && action == flipAction {
+				(*subjectIntersect)[key] = toBeFlippedTo
+
+				continue
+			}
+
+			(*subjectIntersect)[key] = setAction
+		}
+	}
+}
+
+func getRoleBindingListToUpdate(subjectIntersect *map[string]iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType) (blData []*iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDelta) {
+	for k, v := range *subjectIntersect {
+		if v != iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpTypeOPTYPEUNSPECIFIED {
+			values := strings.Split(k, roleSubjectDelimiter)
+
+			blData = append(blData, &iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDelta{
+				Role: values[0],
+				Subject: &iammodel.VmwareTanzuCoreV1alpha1PolicySubject{
+					Name: values[1],
+					Kind: (*iammodel.VmwareTanzuCoreV1alpha1PolicySubjectKind)(&values[2]),
+				},
+				Op: v.Pointer(),
+			})
+		}
+	}
+
+	return blData
+}
+
+func resourceIAMPolicyInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config, _ := m.(authctx.TanzuContext)
+
+	var (
+		UID              string
+		blData           []*iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDelta
+		serverRBL        []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding
+		subjectIntersect = make(map[string]iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType)
+	)
+
+	scopedFullname := constructScope(d)
+	if scopedFullname == nil {
+		return diag.Errorf("unable to update Role Binding; Scope full name is empty")
+	}
+
+	policyList, err := retrieveRoleBindingListFromServer(config, scopedFullname)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, policy := range policyList {
+		if policy.Meta.UID == d.State().ID {
+			serverRBL = append(serverRBL, policy.RoleBindings...)
```

Comment:
if the serverRBL is not used anywhere in this block, then can you invoke 
constructRBOpForUpdate(policy.RoleBindings, &subjectIntersect, iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpTypeOPTYPEUNSPECIFIED)

Reply:
Done